### ISSUE TITLE
inconsistent behavior in multiple-cursor mode

### ISSUE THREAD

**Issue by drammock** [rstudio/rstudio#3291]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.456
    OS Version      : Ubuntu 16.04
    R Version       : 3.4.4

### Steps to reproduce the problem

1. create a new file in the RStudio editor
1. make several blank lines by pressing `Return` a bunch of times
1. create multiple cursors using keyboard shortcut (default is `Ctrl+Alt+Up/Down` I think; mine is customized to `Shift+Alt+Up/Down` but I don't think that matters)
1. try using `esc` or arrow keys (doesn't work)
1. switch focus to another program, then switch back (either via mouse or `Alt-Tab`)
1. now `esc` and arrow keys work with multiple cursors

### Describe the problem in detail

After creating multiple cursors, reverting to a single cursor via `esc` doesn't always work, nor does moving the multiple cursors with the arrow keys.  The steps above are reproducible for me, but I've also seen the behavior on lines that have code on them, both in files that are saved on disk and files with unsaved changes.

### Describe the behavior you expected

when multiple cursors are present, `esc` should revert to single cursor, and arrow keys should move multiple cursors, always.

**Comment by drammock** [comment]
@kevinushey I am skeptical of that explanation.  On my system, pressing and then releasing `Alt` when in the RStudio editor does _not_ leave the menu bar focused.  If I want to access the file menu, for example, I have to press `f` while holding down `Alt`.  Pressing `Alt` and then `f` in sequence just types the letter `f` into the open file in the RStudio editor.

**Comment by kevinushey** [comment]
@drammock I can reproduce with a similar sequence of steps:

1. Press and hold <kbd>Shift</kbd>,
2. Press and hold <kbd>Alt</kbd>,
3. Release <kbd>Shift</kbd>,
4. Release <kbd>Alt</kbd>.

In my case, I see the IDE getting stuck in a state where the arrow keys no longer work in the editor. Other permutations of pressing-and-holding of these modifier keys doesn't seem to reproduce the issue.

**Comment by drammock** [comment]
interesting, I confirm your sequence makes arrow keys unresponsive (even in single-cursor mode).

**Comment by OJWatson** [comment]
Hi, 

Not sure if this helps/is related but seems like this issue extends to impacting all `Shift`+`Alt`+`Arrowkey` keyboard shortcuts within the IDE source window, such that shortcuts that start by pressing `Shift`+`Alt` only work when they are entered `Alt`+`Shift`.

For example the following only work when `Alt` is pressed before `Shift`.
```
Copy Lines Up/Down | Shift+Alt+Up/Down
Select to Line Start/End | Shift+Alt+Left/Right
```
However keyboard shortcuts that are followed by a letter work fine regardless of the order, e.g:
```
New Terminal | Shift+Alt+T
Unfold Selected | Shift+Alt+L
```
Also when going to edit shortcut keys it is not possible to enter one as `Alt`+`Shift`+`<anything>`without it being reversed to `Shift`+`Alt`+`<anything>`. Not sure if it's related or helpful but seemed possibly related. 

### System Info
```
R version: 3.5.1 (2018-07-02)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Linux Mint 19
RStudio Edition: Desktop
RStudio Version: 1.2.1186 (though also had this issue with RStudio 1.1.463 which is why i tried devel.)
```

**Comment by drammock** [comment]
This appears to be fixed in version 1.4.1103 (possibly was fixed earlier, not sure)

**Comment by tongyaop** [comment]
I think I figured this out on Windows:
1. press `ctr` + `shift` + `alt`, use arrow keys, add one extra cursor
2. press `ctr` + `shift` + `alt`, Hold on to `ctr` + `alt`,  but release `shift`, use arrow keys, to add cursor to multiple columns
